### PR TITLE
Add a TryAdd method to RequestHeaders and use it

### PR DESCRIPTION
**Pull request by andreaTP**
Preparation for: https://github.com/microsoft/kiota/pull/3401

### CODE REVIEW

**Review by rkodev — APPROVED** [review]

**Review by rkodev — CHANGES_REQUESTED** [review]
Please add a changelog entry

**Comment by andreaTP** [comment]
done 👍

**Review by rkodev — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+## [1.2.3] - 2023-09-05
```
Comment (rkodev) [inline CHANGELOG.md:14]: Please check the date

Comment (andreaTP) [reply CHANGELOG.md]: fixed, sorry for the noise

**Review by rkodev — APPROVED** [review]

**Pull request merged**